build/check-images.js

```diff
@@ -19,14 +19,27 @@ function checkImageReferences(doc, $, options, { url, rawContent }) {
 
   const checkImages = options.flawLevels.get("images") !== FLAW_LEVELS.IGNORE;
 
+  const checked = new Map();
+
   function addImageFlaw(
     $img,
     src,
     { explanation, externalImage = false, suggestion = null }
   ) {
-    for (const match of findMatchesInText(src, rawContent, {
-      attribute: "src",
-    })) {
+    // If the document has *two* `<img src="XXX">` tags, and assume there's
```

Comment:
I had to read more of the surrounding code to get it. `addImageFlaw` is called *two* times if there are *two* `<img src="XXX"> tags. For me this comment tells me that it's only called once. But maybe I'm misreading this.

Reply:
I rewrote the comment. I tried to really explain it "from the heart" so that it would make the most sense.